Kernel/Tasks/PerformanceEventBuffer.h

```diff
@@ -145,7 +206,7 @@ class PerformanceEventBuffer {
     size_t m_count { 0 };
     NonnullOwnPtr<KBuffer> m_buffer;
 
-    HashMap<NonnullOwnPtr<KString>, size_t> m_strings;
+    SpinlockProtected<HashMap<NonnullOwnPtr<KString>, size_t>, LockRank::None> m_strings;
```

Comment:
Adding a lock around this is a significant change on its own. Can you split that into a separate commit?

Reply:
I have added a separate [commit](https://github.com/SerenityOS/serenity/pull/19950/commits/904bd1a1609c0572974678afd10562ec2b948143) as requested.